src/Kernel.php

```diff
@@ -164,4 +156,30 @@ private function loadLocalConfiguration(YamlFileLoader $loader): void
             $loader->load($this->localConfigFilePath);
         }
     }
+
+    /**
+     * Logs an exception. If this is an \Amp\MultiReasonException all reasons are logged
+     * @param \Throwable $exception
+     * @param LoggerInterface $logger
+     */
+    private function logException(\Throwable $exception, LoggerInterface $logger): void
+    {
+        if ($exception instanceof MultiReasonException) {
+            foreach ($exception->getReasons() as $reason) {
+                $this->logException($reason, $logger);
+            }
+            return;
+        }
```

Comment:
I think it's better to have a single log entry even in such case. If someone uses monolog's mail handler he will get an  email for every reason. I'd prefer to have a single log entry with all the reasons in the log context.
We should also add an integration test about this.

Reply:
Good point about the one single log entry! I only checked the console for now because we're still in developing phase. Refactored to using the log context as suggested.

- Created integration test
- Updated already existing integration test as well (it never reached the line 'hasCriticalThatContains' because of the thrown exception, and 'hasCriticalThatContains' does no assertion on itself)